File: swea/2001_파리퇴치.py
# import sys
# sys.stdin = open('2001ans.txt', 'r')

def pari(i, j):
    pariSum = 0
    for verti in range(M):
        for hori in range(M):
            pariSum += arr[i+verti][j+hori]
    return pariSum

T = int(input())
for tc in range(1, T+1):
    N, M = map(int, input().split())
    arr = [list(map(int, input().split())) for _ in range(N)]

    sum = 0
    for i in range(N-M+1):
        for j in range(N-M+1):
            pariSum = pari(i, j)
            if pariSum > sum:
                sum = pariSum

    print('#{} {}'.format(tc, sum))

File: swea/2001_파리퇴치.java
// 리팩터링 코드 => 메서드 만들고 불러오는 코드로도 만들어보기
import java.util.Scanner;
import java.util.Arrays;

public class Solution {
	public static void main(String[] args) {
		Scanner input = new Scanner(System.in);
		int T = input.nextInt();
		
		for (int tc = 0; tc < T; tc++) {
			int N = input.nextInt();
			int M = input.nextInt();
			int[][] arr = new int[N][N];
			
			for (int i=0; i<N; i++) {
				for (int j=0; j<N; j++) {
					arr[i][j] = input.nextInt();
				}
			}
			
			int max = 0;
			
			for (int i=0; i<N-M+1; i++) {
				for (int j=0; j<N-M+1; j++) {
					int pariSum = 0;
					for (int verti = 0; verti < M; verti++) {
						for (int hori = 0; hori < M; hori++) {
							pariSum += arr[i + verti][j + hori];
						}
					}
					if (pariSum > max) {
						max = pariSum;
					}
				}
			}
		System.out.printf("#%d %d", tc+1, max);
		System.out.println("");
			
		}
	}
}
